File: examples/pokemon/python/pokemon_example.py
"""
PokeAPI Example - Get Pokemon Data
Documentation: https://pokeapi.co/docs/v2
"""
import requests

def get_pokemon(name_or_id):
    """
    Fetch Pokemon data by name or ID
    
    Args:
        name_or_id: Pokemon name (e.g., 'pikachu') or ID (e.g., 25)
    
    Returns:
        dict: Pokemon data including name, types, abilities, stats
    """
    try:
        # Make API request (no auth required!)
        response = requests.get(f'https://pokeapi.co/api/v2/pokemon/{name_or_id}')
        response.raise_for_status()  # Raise error for bad status codes
        
        # Parse JSON response
        data = response.json()
        
        # Extract useful information
        pokemon_info = {
            'name': data['name'].capitalize(),
            'id': data['id'],
            'height': data['height'] / 10,  # Convert to meters
            'weight': data['weight'] / 10,  # Convert to kg
            'types': [t['type']['name'] for t in data['types']],
            'abilities': [a['ability']['name'] for a in data['abilities']],
            'base_stats': {stat['stat']['name']: stat['base_stat'] 
                           for stat in data['stats']}
        }
        
        return pokemon_info
        
    except requests.exceptions.HTTPError as e:
        if e.response.status_code == 404:
            print(f"❌ Pokemon '{name_or_id}' not found!")
        else:
            print(f"❌ HTTP Error: {e}")
        return None
    except requests.exceptions.RequestException as e:
        print(f"❌ Request failed: {e}")
        return None

# Example usage
if __name__ == '__main__':
    # Get Pikachu's data
    pikachu = get_pokemon('pikachu')
    
    if pikachu:
        print(f"\n⚡ {pikachu['name']} (#{pikachu['id']})")
        print(f"📏 Height: {pikachu['height']}m")
        print(f"⚖️  Weight: {pikachu['weight']}kg")
        print(f"🏷️  Types: {', '.join(pikachu['types'])}")
        print(f"💪 Abilities: {', '.join(pikachu['abilities'])}")
        print(f"📊 Base Stats:")
        for stat, value in pikachu['base_stats'].items():
            print(f"   - {stat}: {value}")
    
    # Try with Pokemon ID
    print("\n" + "="*50)
    charizard = get_pokemon(6)  # Charizard's ID
    if charizard:
        print(f"\n🔥 {charizard['name']} (#{charizard['id']})")
        print(f"🏷️  Types: {', '.join(charizard['types'])}")

File: examples/pokemon/javascript/pokemon_example.js
/**
 * PokeAPI Example - Get Pokemon Data
 * Documentation: https://pokeapi.co/docs/v2
 * Works in both Node.js and Browser
 */

/**
 * Fetch Pokemon data by name or ID
 * @param {string|number} nameOrId - Pokemon name (e.g., 'pikachu') or ID (e.g., 25)
 * @returns {Promise<object>} Pokemon data
 */
async function getPokemon(nameOrId) {
    try {
        // Make API request (no auth required!)
        const response = await fetch(`https://pokeapi.co/api/v2/pokemon/${nameOrId}`);

        if (!response.ok) {
            if (response.status === 404) {
                console.error(`❌ Pokemon '${nameOrId}' not found!`);
                return null;
            }
            throw new Error(`HTTP error! status: ${response.status}`);
        }

        // Parse JSON response
        const data = await response.json();

        // Extract useful information
        const pokemonInfo = {
            name: data.name.charAt(0).toUpperCase() + data.name.slice(1),
            id: data.id,
            height: data.height / 10,  // Convert to meters
            weight: data.weight / 10,  // Convert to kg
            types: data.types.map(t => t.type.name),
            abilities: data.abilities.map(a => a.ability.name),
            baseStats: Object.fromEntries(
                data.stats.map(stat => [stat.stat.name, stat.base_stat])
            )
        };

        return pokemonInfo;

    } catch (error) {
        console.error('❌ Request failed:', error.message);
        return null;
    }
}

// Example usage
(async () => {
    // Get Pikachu's data
    const pikachu = await getPokemon('pikachu');

    if (pikachu) {
        console.log(`\n⚡ ${pikachu.name} (#${pikachu.id})`);
        console.log(`📏 Height: ${pikachu.height}m`);
        console.log(`⚖️  Weight: ${pikachu.weight}kg`);
        console.log(`🏷️  Types: ${pikachu.types.join(', ')}`);
        console.log(`💪 Abilities: ${pikachu.abilities.join(', ')}`);
        console.log(`📊 Base Stats:`);
        for (const [stat, value] of Object.entries(pikachu.baseStats)) {
            console.log(`   - ${stat}: ${value}`);
        }
    }

    // Try with Pokemon ID
    console.log('\n' + '='.repeat(50));
    const charizard = await getPokemon(6);  // Charizard's ID
    if (charizard) {
        console.log(`\n🔥 ${charizard.name} (#${charizard.id})`);
        console.log(`🏷️  Types: ${charizard.types.join(', ')}`);
    }
})();

// For Node.js, you can also export the function
if (typeof module !== 'undefined' && module.exports) {
    module.exports = { getPokemon };
}
